app/scripts/metamask-controller.js

```diff
@@ -1284,14 +1307,16 @@ export default class MetamaskController extends EventEmitter {
   async fetchInfoToSync() {
     // Preferences
     const {
-      accountTokens,
       currentLocale,
       frequentRpcList,
       identities,
       selectedAddress,
-      tokens,
     } = this.preferencesController.store.getState();
 
+    const { tokens, allTokens } = this.tokensController.state;
+
+    const accountTokens = allTokens[selectedAddress];
+
     // Filter ERC20 tokens
     const filteredAccountTokens = {};
     Object.keys(accountTokens).forEach((address) => {
```

Comment:
Confirm whether or not we need this

Reply:
So I'm fairly certain we still want this. We need to filter out the non-ERC20 tokens because they are handled differently by mobile. We don't have a great way to keep pass say ERC721 tokens through the sync and I think we might as well wait until we have better handling of them in the extension to include them as part of the sync. 

I've altered the logic here to do a more thorough sweep to filter out ERC721 tokens. Please take another look!